### ISSUE TITLE
Add cleanup and upsert verification to the transport tests

### ISSUE THREAD

**Issue by prestonvasquez** [alpstable/gidari#204]
The transport integration tests in internal/transport/transport_test.go do not verify that the upserts actually occur. We need to do something similar to what we do in storage and [truncate the tables on cleanup](https://github.com/alpine-hodler/gidari/blob/main/internal/storage/storage_test.go#L81) and also [verify that the data was actually inserted](https://github.com/alpine-hodler/gidari/blob/main/internal/storage/storage_test.go#L128).

To use the `ListTables` method with a repository, see [this](https://github.com/alpine-hodler/gidari/blob/main/examples/repository/example_test.go#L172) example.

**Comment by ash-kamrip** [comment]
hello @prestonvasquez I would like to take this up ! , just wanted to ask few things though, the `stg` is created against each dns string in the examples, in `upsert` we are going to get the dns from cfg.connectionstrings and then get the tables and verify if they were populated. I don't understand why we truncate the tables on cleanup/ or why do we use them at all ? I am a beginner so apologies if i this is something commonly understood

**Comment by prestonvasquez** [comment]
@ash-kamrip The specific test that we want to add this logic to is [this](https://github.com/alpine-hodler/gidari/blob/main/internal/transport/transport_test.go#L184) one. 

For each configuration file in the [testdata](https://github.com/alpine-hodler/gidari/tree/main/internal/transport/testdata/upsert) directory, the `transport.Upsert` method is called by that integration test. When  Upsert is called the `Config.repos` method will be called and create a connection to the MongoDB server. For example, when the [pokemon.yml file](https://github.com/alpine-hodler/gidari/blob/main/internal/transport/testdata/upsert/pokemon.yml#L8) is upserted the `Config.repos` method will be called and create a connection to the MongoDB server. Once the we fetch the payload for the `/api/v2/pokemon/ditto` endpoint it will be inserted into a collection on that server called `ditto`. 

The testing algorithm should look something like this:

```
func TestUpsert {
  FOR configs IN testdata {
      UPSERT config 
      
      // Make sure all the data we inserted in the integration test has been destroyed 
      // after the test has ended
      DEFER truncateTable
      
      // If no data was inserted, then the test upsert didn't work and we should throw a fatal.
      IF tableSize == bytes(0) {
          THROW FATAL
      }
  
}
```

Does this make sense?

**Comment by prestonvasquez** [comment]
Closed by #360
